Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py

```diff
@@ -4667,6 +4688,488 @@ def apply_quarantine_file_action_command(args: dict) -> CommandResults:
     )
 
 
+def build_cs_falcon_filter(custom_filter: str | None = None, **filter_args) -> str:
+    """Creates an FQL syntax filter from a dictionary and a custom built filter
+
+    :custom_filter: custom filter from user (will take priority if conflicts with dictionary), defaults to None
+    :filter_args: args to translate to FQL format.
+
+    :return: FQL syntax filter.
+    """
+
+    custom_filter_list = custom_filter.split('+') if custom_filter else []
+    arguments = [f'{key}:{argToList(value)}' for key, value in filter_args.items() if value]
+    # custom_filter takes priority because it is first
+    return "%2B".join(custom_filter_list + arguments)
+
+
+def ODS_query_scans_request(**query_params) -> dict:
+
+    remove_nulls_from_dictionary(query_params)
+    # http_request messes up the params, so they were put directly in the url:
+    url_params = "&".join(f"{k}={v}" for k, v in query_params.items())
+    return http_request('GET', f'/ods/queries/scans/v1?{url_params}')
+
+
+def ODS_get_scans_by_id_request(ids: list[str]) -> dict:
+
+    url_params = '&'.join(f'ids={query_id}' for query_id in ids)
+    return http_request('GET', f'/ods/entities/scans/v1?{url_params}')
+
+
+def map_scan_resource_to_UI(resource: dict) -> dict:
+    output = {
+        'ID': resource.get('id'),
+        'Status': resource.get('status'),
+        'Severity': resource.get('severity'),
+        'File Count': '\n-\n'.join('\n'.join(f'{k}: {v}' for k, v in filecount.items())
```

Comment:
why is filecount not a number? What is the output here? Please add docstrings

Reply:
np